proposals/20210605-schema_curation_custom_component.md

```diff
@@ -0,0 +1,37 @@
+#### SIG TFX-Addons
+# Project Proposal
+
+**Your name:** Pratishtha Abrol
+
+**Your email:** pratishthaabrol@gmail.com
+
+**Your company/organization:** Outreachy
+
+**Project name:** [Schema curation custom component](https://github.com/tensorflow/tfx-addons/issues/8)
+
+## Project Description
+This project applies Python user code from a user-supplied module file to a schema produced by SchemaGen, to curate the schema based on domain knowledge.
+
+## Project Category
+Component
+
+## Project Use-Case(s)
+This project will allow the user to add a custom component that modifies the schema generated by SchemaGen component according to user knowledge, for example, fixing domain limits that were inferred wrongly by the SchemaGen component.
+
+## Project Implementation
+Implementation of the Schema Curation Custom Component can be done using the following approach:
+- Get the base Schema using SchemaGen component of TFX
+- User supplies a module file with a fully-custom component that defines the additions/changes to the initially generated schema through SchemaGen.
+- And execution script would run on the module file, which sets and modifies variables accordingly.
+- The base schema gets modified according to the module file and used further along the pipeline
+
+## Project Dependencies
```

Comment:
It's probably a good idea to include [Transform](https://github.com/tensorflow/transform) as a dependency.  The [schema_utils](https://github.com/tensorflow/transform/blob/master/tensorflow_transform/tf_metadata/schema_utils.py) will be useful.

Reply:
yes sir, will do!